developer-portal-wrapper/developer_portal_wrapper.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env python3
+"""
+Run push-cgw-metadata command to push metadata to CGW:
+1. Extract all components under contentGateway key from dataPath
+2. Find all the files in contentDir that starts with the component name
+4. Generate necessary metadata for each file
+5. Dump the metadata to a YAML file
+6. Run push-cgw-metadata to push the metadata
+"""
+import os
+import yaml
+import hashlib
+import subprocess
+import argparse
+import logging
+
+LOG = logging.getLogger("developer-portal-wrapper")
+DEFAULT_LOG_FMT = "%(asctime)s [%(levelname)-8s] %(message)s"
+DEFAULT_DATE_FMT = "%Y-%m-%d %H:%M:%S %z"
+CGW_ENV_VARS_STRICT = (
+    "CGW_USERNAME",
+    "CGW_TOKEN",
+)
+
+WORKSPACE_DIR = "/tmp"
+METADATA_FILE_PATH = f"{WORKSPACE_DIR}/cgw_metadata.yaml"
+RESULT_FILE_JSON_PATH = f"{WORKSPACE_DIR}/results.json"
+
+os.makedirs(WORKSPACE_DIR, exist_ok=True)
+
+
+def generate_download_url(content_dir, file_name):
+    """
+    Generate a download URL in this format:
+    /content/origin/files/sha256/{checksum[:2]}{checksum}/{file_name}
+    """
+    prefix = "/content/origin/files/sha256"
+    sha256_hash = hashlib.sha256()
+    with open(content_dir + "/" + file_name, "rb") as f:
+        for byte_block in iter(lambda: f.read(4096), b""):
+            sha256_hash.update(byte_block)
+    checksum = sha256_hash.hexdigest()
+    return f"{prefix}/{checksum[:2]}/{checksum}/{file_name}"
+
+
+def generate_metadata(
+    product_name, product_code, product_version_name, content_dir, content_list, file_prefix
+):
+    """
+    Generate metadata for each file in
+    content_list that starts with the component name
+    """
+    metadata = []
+
+    default_values_per_component = {
+        "type": "FILE",
+        "shortURL": f"/cgw/{product_code}",
+        "hidden": False,
+        "invisible": False,
+    }
+
+    for file in content_list:
+        matching_component = file.startswith(file_prefix)
+
+        if matching_component:
+            LOG.info("Processing file: %s", file)
+            component_item = {
+                "productName": product_name,
+                "productCode": product_code,
+                "productVersionName": product_version_name,
+                "downloadURL": generate_download_url(content_dir, file),
+                "label": file,
+            }
+            default_values_per_component["shortURL"] += f"/{file}"
+            metadata.append(
+                {
+                    "type": "file",
+                    "action": "create",
+                    "metadata": {**default_values_per_component, **component_item},
+                }
+            )
+        else:
+            LOG.warning(f"Skipping file: {file} as it does not start with any component name")
+
+    return metadata
+
+
+def validate_args(args):
+    assert all([os.getenv(item) for item in CGW_ENV_VARS_STRICT]), (
+        f"Provide all required CGW environment variables: " f"{', '.join(CGW_ENV_VARS_STRICT)}"
```

Comment:
Why split the string in two if it fits on one line?
```suggestion
        f"Provide all required CGW environment variables: {', '.join(CGW_ENV_VARS_STRICT)}"
```

Reply:
This was copy-paste...I fixed it per your suggestion.